From: Steve Markgraf <redacted>
Date: Wed, 17 Jan 2024 19:18:34 +0000 (+0100)
Subject: fx2lib: fix SDCC warning 283
X-Git-Url: http://sigrok.org/gitweb/?p=sigrok-firmware-fx2lafw.git;a=commitdiff_plain;h=bb3c6dfdb3f95bf529ed95dcc5fb3610e4473673

fx2lib: fix SDCC warning 283

This fixes the following warnings that appear with
SDCC >= 4.3.0:

"warning 283: function declarator with no prototype"
---

diff --git a/fx2lib/include/autovector.h b/fx2lib/include/autovector.h
index 38f80fcd..ecbeb411 100644
--- a/fx2lib/include/autovector.h
+++ b/fx2lib/include/autovector.h
@@ -172,49 +172,49 @@ typedef enum {
 // you must include the predef of these in the file with your main
 // so lets just define them here
 
-void sudav_isr() __interrupt(SUDAV_ISR);
-void sof_isr() __interrupt(SOF_ISR);
-void sutok_isr() __interrupt(SUTOK_ISR);
-void suspend_isr() __interrupt(SUSPEND_ISR);
-void usbreset_isr() __interrupt(USBRESET_ISR);
-void hispeed_isr() __interrupt(HISPEED_ISR);
-void ep0ack_isr() __interrupt(EP0ACK_ISR);
-void ep0in_isr() __interrupt(EP0IN_ISR);
-void ep0out_isr() __interrupt(EP0OUT_ISR);
-void ep1in_isr() __interrupt(EP1IN_ISR);
-void ep1out_isr() __interrupt(EP1OUT_ISR);
-void ep2_isr() __interrupt(EP2_ISR);
-void ep4_isr() __interrupt(EP4_ISR);
-void ep6_isr() __interrupt(EP6_ISR);
-void ep8_isr() __interrupt(EP8_ISR);
-void ibn_isr() __interrupt(IBN_ISR);
-void ep0ping_isr() __interrupt(EP0PING_ISR);
-void ep1ping_isr() __interrupt(EP1PING_ISR);
-void ep2ping_isr() __interrupt(EP2PING_ISR);
-void ep4ping_isr() __interrupt(EP4PING_ISR);
-void ep6ping_isr() __interrupt(EP6PING_ISR);
-void ep8ping_isr() __interrupt(EP8PING_ISR);
-void errlimit_isr() __interrupt(ERRLIMIT_ISR);
-void ep2isoerr_isr() __interrupt(EP2ISOERR_ISR);
-void ep4isoerr_isr() __interrupt(EP4ISOERR_ISR);
-void ep6isoerr_isr() __interrupt(EP6ISOERR_ISR);
-void ep8isoerr_isr() __interrupt(EP8ISOERR_ISR);
-void spare_isr() __interrupt(RESERVED_ISR); // not used
+void sudav_isr(void) __interrupt(SUDAV_ISR);
+void sof_isr(void) __interrupt(SOF_ISR);
+void sutok_isr(void) __interrupt(SUTOK_ISR);
+void suspend_isr(void) __interrupt(SUSPEND_ISR);
+void usbreset_isr(void) __interrupt(USBRESET_ISR);
+void hispeed_isr(void) __interrupt(HISPEED_ISR);
+void ep0ack_isr(void) __interrupt(EP0ACK_ISR);
+void ep0in_isr(void) __interrupt(EP0IN_ISR);
+void ep0out_isr(void) __interrupt(EP0OUT_ISR);
+void ep1in_isr(void) __interrupt(EP1IN_ISR);
+void ep1out_isr(void) __interrupt(EP1OUT_ISR);
+void ep2_isr(void) __interrupt(EP2_ISR);
+void ep4_isr(void) __interrupt(EP4_ISR);
+void ep6_isr(void) __interrupt(EP6_ISR);
+void ep8_isr(void) __interrupt(EP8_ISR);
+void ibn_isr(void) __interrupt(IBN_ISR);
+void ep0ping_isr(void) __interrupt(EP0PING_ISR);
+void ep1ping_isr(void) __interrupt(EP1PING_ISR);
+void ep2ping_isr(void) __interrupt(EP2PING_ISR);
+void ep4ping_isr(void) __interrupt(EP4PING_ISR);
+void ep6ping_isr(void) __interrupt(EP6PING_ISR);
+void ep8ping_isr(void) __interrupt(EP8PING_ISR);
+void errlimit_isr(void) __interrupt(ERRLIMIT_ISR);
+void ep2isoerr_isr(void) __interrupt(EP2ISOERR_ISR);
+void ep4isoerr_isr(void) __interrupt(EP4ISOERR_ISR);
+void ep6isoerr_isr(void) __interrupt(EP6ISOERR_ISR);
+void ep8isoerr_isr(void) __interrupt(EP8ISOERR_ISR);
+void spare_isr(void) __interrupt(RESERVED_ISR); // not used
 // gpif ints
-void ep2pf_isr() __interrupt(EP2PF_ISR);
-void ep4pf_isr() __interrupt(EP4PF_ISR);
-void ep6pf_isr() __interrupt(EP6PF_ISR);
-void ep8pf_isr() __interrupt(EP8PF_ISR);
-void ep2ef_isr() __interrupt(EP2EF_ISR);
-void ep4ef_isr() __interrupt(EP4EF_ISR);
-void ep6ef_isr() __interrupt(EP6EF_ISR);
-void ep8ef_isr() __interrupt(EP8EF_ISR);
-void ep2ff_isr() __interrupt(EP2FF_ISR);
-void ep4ff_isr() __interrupt(EP4FF_ISR);
-void ep6ff_isr() __interrupt(EP6FF_ISR);
-void ep8ff_isr() __interrupt(EP8FF_ISR);
-void gpifdone_isr() __interrupt(GPIFDONE_ISR);
-void gpifwf_isr() __interrupt(GPIFWF_ISR);
+void ep2pf_isr(void) __interrupt(EP2PF_ISR);
+void ep4pf_isr(void) __interrupt(EP4PF_ISR);
+void ep6pf_isr(void) __interrupt(EP6PF_ISR);
+void ep8pf_isr(void) __interrupt(EP8PF_ISR);
+void ep2ef_isr(void) __interrupt(EP2EF_ISR);
+void ep4ef_isr(void) __interrupt(EP4EF_ISR);
+void ep6ef_isr(void) __interrupt(EP6EF_ISR);
+void ep8ef_isr(void) __interrupt(EP8EF_ISR);
+void ep2ff_isr(void) __interrupt(EP2FF_ISR);
+void ep4ff_isr(void) __interrupt(EP4FF_ISR);
+void ep6ff_isr(void) __interrupt(EP6FF_ISR);
+void ep8ff_isr(void) __interrupt(EP8FF_ISR);
+void gpifdone_isr(void) __interrupt(GPIFDONE_ISR);
+void gpifwf_isr(void) __interrupt(GPIFWF_ISR);
 
 #endif
 
diff --git a/fx2lib/include/serial.h b/fx2lib/include/serial.h
index 7c5935eb..c83959c4 100644
--- a/fx2lib/include/serial.h
+++ b/fx2lib/include/serial.h
@@ -51,4 +51,4 @@ void sio0_init( WORD baud_rate ) __critical ; // baud_rate max should be 57600 s
  Just use one or the other. (This makes terminal echo easy)
 **/
 void putchar(char c);
-char getchar();
+char getchar(void);
diff --git a/fx2lib/include/setupdat.h b/fx2lib/include/setupdat.h
index 5dba882a..fc9e40c6 100644
--- a/fx2lib/include/setupdat.h
+++ b/fx2lib/include/setupdat.h
@@ -148,7 +148,7 @@ __xdata BYTE* ep_addr(BYTE ep);
  You can also use the usbjt and enable the sudav isr
  and call the function from withing the sudav isr routine
 */
-void handle_setupdata();
+void handle_setupdata(void);
 
 
 /**
diff --git a/fx2lib/lib/interrupts/ep0ack_isr.c b/fx2lib/lib/interrupts/ep0ack_isr.c
index d216717d..547e0bfc 100644
--- a/fx2lib/lib/interrupts/ep0ack_isr.c
+++ b/fx2lib/lib/interrupts/ep0ack_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0ack_isr() __interrupt(EP0ACK_ISR) {}
+void ep0ack_isr(void) __interrupt(EP0ACK_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0in_isr.c b/fx2lib/lib/interrupts/ep0in_isr.c
index f3d8990e..c55d0fbc 100644
--- a/fx2lib/lib/interrupts/ep0in_isr.c
+++ b/fx2lib/lib/interrupts/ep0in_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0in_isr() __interrupt(EP0IN_ISR) {}
+void ep0in_isr(void) __interrupt(EP0IN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0out_isr.c b/fx2lib/lib/interrupts/ep0out_isr.c
index a5acedb7..a93cd77c 100644
--- a/fx2lib/lib/interrupts/ep0out_isr.c
+++ b/fx2lib/lib/interrupts/ep0out_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0out_isr() __interrupt(EP0OUT_ISR) {}
+void ep0out_isr(void) __interrupt(EP0OUT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0ping_isr.c b/fx2lib/lib/interrupts/ep0ping_isr.c
index 7b1ee929..218b7528 100644
--- a/fx2lib/lib/interrupts/ep0ping_isr.c
+++ b/fx2lib/lib/interrupts/ep0ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0ping_isr() __interrupt(EP0PING_ISR) {}
+void ep0ping_isr(void) __interrupt(EP0PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1in_isr.c b/fx2lib/lib/interrupts/ep1in_isr.c
index b666fcb3..4197bd54 100644
--- a/fx2lib/lib/interrupts/ep1in_isr.c
+++ b/fx2lib/lib/interrupts/ep1in_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1in_isr() __interrupt(EP1IN_ISR) {}
+void ep1in_isr(void) __interrupt(EP1IN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1out_isr.c b/fx2lib/lib/interrupts/ep1out_isr.c
index 6c87dc20..8e19ca01 100644
--- a/fx2lib/lib/interrupts/ep1out_isr.c
+++ b/fx2lib/lib/interrupts/ep1out_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1out_isr() __interrupt(EP1OUT_ISR) {}
+void ep1out_isr(void) __interrupt(EP1OUT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1ping_isr.c b/fx2lib/lib/interrupts/ep1ping_isr.c
index c32d437c..62849214 100644
--- a/fx2lib/lib/interrupts/ep1ping_isr.c
+++ b/fx2lib/lib/interrupts/ep1ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1ping_isr() __interrupt(EP1PING_ISR) {}
+void ep1ping_isr(void) __interrupt(EP1PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2_isr.c b/fx2lib/lib/interrupts/ep2_isr.c
index 4de637a3..4596b3ce 100644
--- a/fx2lib/lib/interrupts/ep2_isr.c
+++ b/fx2lib/lib/interrupts/ep2_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2_isr() __interrupt(EP2_ISR) {}
+void ep2_isr(void) __interrupt(EP2_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2ef_isr.c b/fx2lib/lib/interrupts/ep2ef_isr.c
index 46f3f87a..a88f1571 100644
--- a/fx2lib/lib/interrupts/ep2ef_isr.c
+++ b/fx2lib/lib/interrupts/ep2ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ef_isr() __interrupt(EP2EF_ISR){}
+void ep2ef_isr(void) __interrupt(EP2EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2ff_isr.c b/fx2lib/lib/interrupts/ep2ff_isr.c
index 400e4803..1c073fbd 100644
--- a/fx2lib/lib/interrupts/ep2ff_isr.c
+++ b/fx2lib/lib/interrupts/ep2ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ff_isr() __interrupt(EP2FF_ISR){}
+void ep2ff_isr(void) __interrupt(EP2FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2isoerr_isr.c b/fx2lib/lib/interrupts/ep2isoerr_isr.c
index d0ca4a89..7b121a51 100644
--- a/fx2lib/lib/interrupts/ep2isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep2isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2isoerr_isr() __interrupt(EP2ISOERR_ISR) {}
+void ep2isoerr_isr(void) __interrupt(EP2ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2pf_isr.c b/fx2lib/lib/interrupts/ep2pf_isr.c
index 9bd8e9aa..4aa71d33 100644
--- a/fx2lib/lib/interrupts/ep2pf_isr.c
+++ b/fx2lib/lib/interrupts/ep2pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2pf_isr() __interrupt(EP2PF_ISR){}
+void ep2pf_isr(void) __interrupt(EP2PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2ping_isr.c b/fx2lib/lib/interrupts/ep2ping_isr.c
index dad1e953..336616cd 100644
--- a/fx2lib/lib/interrupts/ep2ping_isr.c
+++ b/fx2lib/lib/interrupts/ep2ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ping_isr() __interrupt(EP2PING_ISR) {}
+void ep2ping_isr(void) __interrupt(EP2PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4_isr.c b/fx2lib/lib/interrupts/ep4_isr.c
index 5affac9a..19beb727 100644
--- a/fx2lib/lib/interrupts/ep4_isr.c
+++ b/fx2lib/lib/interrupts/ep4_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4_isr() __interrupt(EP4_ISR) {}
+void ep4_isr(void) __interrupt(EP4_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4ef_isr.c b/fx2lib/lib/interrupts/ep4ef_isr.c
index 4a0cff1a..77c5576b 100644
--- a/fx2lib/lib/interrupts/ep4ef_isr.c
+++ b/fx2lib/lib/interrupts/ep4ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ef_isr() __interrupt(EP4EF_ISR){}
+void ep4ef_isr(void) __interrupt(EP4EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4ff_isr.c b/fx2lib/lib/interrupts/ep4ff_isr.c
index 3278da2d..cbe0b686 100644
--- a/fx2lib/lib/interrupts/ep4ff_isr.c
+++ b/fx2lib/lib/interrupts/ep4ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ff_isr() __interrupt(EP4FF_ISR){}
+void ep4ff_isr(void) __interrupt(EP4FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4isoerr_isr.c b/fx2lib/lib/interrupts/ep4isoerr_isr.c
index dcd7ff90..d198f3b8 100644
--- a/fx2lib/lib/interrupts/ep4isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep4isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4isoerr_isr() __interrupt(EP4ISOERR_ISR) {}
+void ep4isoerr_isr(void) __interrupt(EP4ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4pf_isr.c b/fx2lib/lib/interrupts/ep4pf_isr.c
index 7bbe9f02..14ec7e5b 100644
--- a/fx2lib/lib/interrupts/ep4pf_isr.c
+++ b/fx2lib/lib/interrupts/ep4pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4pf_isr() __interrupt(EP4PF_ISR){}
+void ep4pf_isr(void) __interrupt(EP4PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4ping_isr.c b/fx2lib/lib/interrupts/ep4ping_isr.c
index 8d226856..47de3721 100644
--- a/fx2lib/lib/interrupts/ep4ping_isr.c
+++ b/fx2lib/lib/interrupts/ep4ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ping_isr() __interrupt(EP4PING_ISR) {}
+void ep4ping_isr(void) __interrupt(EP4PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6_isr.c b/fx2lib/lib/interrupts/ep6_isr.c
index 630f9b26..61801f2f 100644
--- a/fx2lib/lib/interrupts/ep6_isr.c
+++ b/fx2lib/lib/interrupts/ep6_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6_isr() __interrupt(EP6_ISR) {}
+void ep6_isr(void) __interrupt(EP6_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6ef_isr.c b/fx2lib/lib/interrupts/ep6ef_isr.c
index e9ba6d54..28b2db39 100644
--- a/fx2lib/lib/interrupts/ep6ef_isr.c
+++ b/fx2lib/lib/interrupts/ep6ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ef_isr() __interrupt(EP6EF_ISR){}
+void ep6ef_isr(void) __interrupt(EP6EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6ff_isr.c b/fx2lib/lib/interrupts/ep6ff_isr.c
index 011fe871..051b6e29 100644
--- a/fx2lib/lib/interrupts/ep6ff_isr.c
+++ b/fx2lib/lib/interrupts/ep6ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ff_isr() __interrupt(EP6FF_ISR){}
+void ep6ff_isr(void) __interrupt(EP6FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6isoerr_isr.c b/fx2lib/lib/interrupts/ep6isoerr_isr.c
index 8c005777..efae2cfd 100644
--- a/fx2lib/lib/interrupts/ep6isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep6isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6isoerr_isr() __interrupt(EP6ISOERR_ISR) {}
+void ep6isoerr_isr(void) __interrupt(EP6ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6pf_isr.c b/fx2lib/lib/interrupts/ep6pf_isr.c
index 52f6d738..dee40231 100644
--- a/fx2lib/lib/interrupts/ep6pf_isr.c
+++ b/fx2lib/lib/interrupts/ep6pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6pf_isr() __interrupt(EP6PF_ISR){}
+void ep6pf_isr(void) __interrupt(EP6PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6ping_isr.c b/fx2lib/lib/interrupts/ep6ping_isr.c
index 5430f209..9ed855f7 100644
--- a/fx2lib/lib/interrupts/ep6ping_isr.c
+++ b/fx2lib/lib/interrupts/ep6ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ping_isr() __interrupt(EP6PING_ISR) {}
+void ep6ping_isr(void) __interrupt(EP6PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8_isr.c b/fx2lib/lib/interrupts/ep8_isr.c
index d415a3ec..8ebce9b8 100644
--- a/fx2lib/lib/interrupts/ep8_isr.c
+++ b/fx2lib/lib/interrupts/ep8_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8_isr() __interrupt(EP8_ISR) {}
+void ep8_isr(void) __interrupt(EP8_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8ef_isr.c b/fx2lib/lib/interrupts/ep8ef_isr.c
index d04d2376..508f49b9 100644
--- a/fx2lib/lib/interrupts/ep8ef_isr.c
+++ b/fx2lib/lib/interrupts/ep8ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ef_isr() __interrupt(EP8EF_ISR){}
+void ep8ef_isr(void) __interrupt(EP8EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8ff_isr.c b/fx2lib/lib/interrupts/ep8ff_isr.c
index d1e5fcd7..88811d7d 100644
--- a/fx2lib/lib/interrupts/ep8ff_isr.c
+++ b/fx2lib/lib/interrupts/ep8ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ff_isr() __interrupt(EP8FF_ISR){}
+void ep8ff_isr(void) __interrupt(EP8FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8isoerr_isr.c b/fx2lib/lib/interrupts/ep8isoerr_isr.c
index a7e8e0f9..fa5f54c5 100644
--- a/fx2lib/lib/interrupts/ep8isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep8isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8isoerr_isr() __interrupt(EP8ISOERR_ISR) {}
+void ep8isoerr_isr(void) __interrupt(EP8ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8pf_isr.c b/fx2lib/lib/interrupts/ep8pf_isr.c
index 53dcdf27..245fd343 100644
--- a/fx2lib/lib/interrupts/ep8pf_isr.c
+++ b/fx2lib/lib/interrupts/ep8pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8pf_isr() __interrupt(EP8PF_ISR){}
+void ep8pf_isr(void) __interrupt(EP8PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8ping_isr.c b/fx2lib/lib/interrupts/ep8ping_isr.c
index 5cf05dc7..56965e6a 100644
--- a/fx2lib/lib/interrupts/ep8ping_isr.c
+++ b/fx2lib/lib/interrupts/ep8ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ping_isr() __interrupt(EP8PING_ISR) {}
+void ep8ping_isr(void) __interrupt(EP8PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/errlimit_isr.c b/fx2lib/lib/interrupts/errlimit_isr.c
index cf4e1f61..921b1def 100644
--- a/fx2lib/lib/interrupts/errlimit_isr.c
+++ b/fx2lib/lib/interrupts/errlimit_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void errlimit_isr() __interrupt(ERRLIMIT_ISR) {}
+void errlimit_isr(void) __interrupt(ERRLIMIT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/gpifdone_isr.c b/fx2lib/lib/interrupts/gpifdone_isr.c
index ee5ce1f6..03f0bb5f 100644
--- a/fx2lib/lib/interrupts/gpifdone_isr.c
+++ b/fx2lib/lib/interrupts/gpifdone_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void gpifdone_isr() __interrupt(GPIFDONE_ISR){}
+void gpifdone_isr(void) __interrupt(GPIFDONE_ISR){}
 
diff --git a/fx2lib/lib/interrupts/gpifwf_isr.c b/fx2lib/lib/interrupts/gpifwf_isr.c
index e9941541..b220a9fe 100644
--- a/fx2lib/lib/interrupts/gpifwf_isr.c
+++ b/fx2lib/lib/interrupts/gpifwf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void gpifwf_isr() __interrupt(GPIFWF_ISR){}
+void gpifwf_isr(void) __interrupt(GPIFWF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/hispeed_isr.c b/fx2lib/lib/interrupts/hispeed_isr.c
index 9daabc12..dcbd0e4b 100644
--- a/fx2lib/lib/interrupts/hispeed_isr.c
+++ b/fx2lib/lib/interrupts/hispeed_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void hispeed_isr() __interrupt(HISPEED_ISR) {}
+void hispeed_isr(void) __interrupt(HISPEED_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ibn_isr.c b/fx2lib/lib/interrupts/ibn_isr.c
index aa0a7523..97fb05e8 100644
--- a/fx2lib/lib/interrupts/ibn_isr.c
+++ b/fx2lib/lib/interrupts/ibn_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ibn_isr() __interrupt(IBN_ISR) {}
+void ibn_isr(void) __interrupt(IBN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sof_isr.c b/fx2lib/lib/interrupts/sof_isr.c
index 64a409c4..8b0b8654 100644
--- a/fx2lib/lib/interrupts/sof_isr.c
+++ b/fx2lib/lib/interrupts/sof_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sof_isr() __interrupt(SOF_ISR) {}
+void sof_isr(void) __interrupt(SOF_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/spare_isr.c b/fx2lib/lib/interrupts/spare_isr.c
index 3b81fd20..effb7d02 100644
--- a/fx2lib/lib/interrupts/spare_isr.c
+++ b/fx2lib/lib/interrupts/spare_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void spare_isr() __interrupt(RESERVED_ISR) {}
+void spare_isr(void) __interrupt(RESERVED_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sudav_isr.c b/fx2lib/lib/interrupts/sudav_isr.c
index 4d36b59a..d5ed73bf 100644
--- a/fx2lib/lib/interrupts/sudav_isr.c
+++ b/fx2lib/lib/interrupts/sudav_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sudav_isr() __interrupt(SUDAV_ISR) {}
+void sudav_isr(void) __interrupt(SUDAV_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/suspend_isr.c b/fx2lib/lib/interrupts/suspend_isr.c
index cf08b28a..1f471179 100644
--- a/fx2lib/lib/interrupts/suspend_isr.c
+++ b/fx2lib/lib/interrupts/suspend_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void suspend_isr() __interrupt(SUSPEND_ISR) {}
+void suspend_isr(void) __interrupt(SUSPEND_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sutok_isr.c b/fx2lib/lib/interrupts/sutok_isr.c
index 8c7368d2..9c424c67 100644
--- a/fx2lib/lib/interrupts/sutok_isr.c
+++ b/fx2lib/lib/interrupts/sutok_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sutok_isr() __interrupt(SUTOK_ISR) {}
+void sutok_isr(void) __interrupt(SUTOK_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/usbreset_isr.c b/fx2lib/lib/interrupts/usbreset_isr.c
index 8888eede..5a91e080 100644
--- a/fx2lib/lib/interrupts/usbreset_isr.c
+++ b/fx2lib/lib/interrupts/usbreset_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void usbreset_isr() __interrupt(USBRESET_ISR) {}
+void usbreset_isr(void) __interrupt(USBRESET_ISR) {}
 
diff --git a/fx2lib/lib/serial.c b/fx2lib/lib/serial.c
index 3114d580..60eb0187 100644
--- a/fx2lib/lib/serial.c
+++ b/fx2lib/lib/serial.c
@@ -68,7 +68,7 @@ void sio0_init( WORD baud_rate ) __critical { // baud_rate max should be 57600 s
 
 }
 
-char getchar() {
+char getchar(void) {
   char c;
   while (!RI)
     ;  
diff --git a/fx2lib/lib/setupdat.c b/fx2lib/lib/setupdat.c
index 878f7b94..a1eb41d1 100644
--- a/fx2lib/lib/setupdat.c
+++ b/fx2lib/lib/setupdat.c
@@ -34,7 +34,7 @@ extern BOOL handle_vendorcommand(BYTE cmd);
 extern BOOL handle_set_configuration(BYTE cfg);
 extern BOOL handle_get_interface(BYTE ifc, BYTE* alt_ifc);
 extern BOOL handle_set_interface(BYTE ifc,BYTE alt_ifc);
-extern BYTE handle_get_configuration();
+extern BYTE handle_get_configuration(void);
 extern BOOL handle_set_configuration(BYTE cfg);
 extern void handle_reset_ep(BYTE ep);
 
@@ -44,16 +44,16 @@ extern void handle_reset_ep(BYTE ep);
 
 
 //  GET_STATUS,
-BOOL handle_get_status();
+BOOL handle_get_status(void);
 //  CLEAR_FEATURE,
-BOOL handle_clear_feature();
+BOOL handle_clear_feature(void);
   // 0x02 is reserved
 //  SET_FEATURE=0x03,
-BOOL handle_set_feature();
+BOOL handle_set_feature(void);
   // 0x04 is reserved
 //  SET_ADDRESS=0x05, // this is handled by EZ-USB core unless RENUM=0
 //  GET_DESCRIPTOR,
-void handle_get_descriptor();
+void handle_get_descriptor(void);
 //  SET_DESCRIPTOR,
 //  GET_CONFIGURATION, // handled by callback
 //  SET_CONFIGURATION, // handled by callback
@@ -68,7 +68,7 @@ void handle_get_descriptor();
  handshake
 */
 
-void handle_setupdata() {
+void handle_setupdata(void) {
     //printf ( "Handle setupdat: %02x\n", SETUPDAT[1] );
 
     switch ( SETUPDAT[1] ) {
@@ -156,7 +156,7 @@ __xdata BYTE* ep_addr(BYTE ep) { // bit 8 of ep_num is the direction
 volatile BOOL self_powered=FALSE;
 volatile BOOL remote_wakeup_allowed=FALSE;
 
-BOOL handle_get_status() {
+BOOL handle_get_status(void) {
     
     switch ( SETUPDAT[0] ) {
 
@@ -201,7 +201,7 @@ BOOL handle_get_status() {
 #define GF_DEVICE 0
 #define GF_ENDPOINT 2
 
-BOOL handle_clear_feature() {
+BOOL handle_clear_feature(void) {
  //printf ( "Clear Feature\n" );
  switch ( SETUPDAT[0] ) {
    case GF_DEVICE:
@@ -227,7 +227,7 @@ BOOL handle_clear_feature() {
  return TRUE;
 }
 
-BOOL handle_set_feature() {
+BOOL handle_set_feature(void) {
  printf ( "Set Feature %02x\n", SETUPDAT[0] );
  switch ( SETUPDAT[0] ) {
   case GF_DEVICE:
@@ -299,7 +299,7 @@ void handle_hispeed(BOOL highspeed) {
  *  String
  *  Other-Speed
  **/
-void handle_get_descriptor() {
+void handle_get_descriptor(void) {
     //printf ( "Get Descriptor\n" );
     
     switch ( SETUPDAT[3] ) {
